cmd/snap-bootstrap/initramfs_systemd_mount.go

```diff
@@ -149,6 +156,11 @@ func doSystemdMountImpl(what, where string, opts *systemdMountOptions) error {
 	if opts.Private {
 		options = append(options, "private")
 	}
+	if len(opts.VerityHashDevice) > 0 && len(opts.VerityRootHash) > 0 {
+		options = append(options, fmt.Sprintf("verity.roothash=%s", opts.VerityRootHash))
+		options = append(options, fmt.Sprintf("verity.hashdevice=%s", opts.VerityHashDevice))
+		options = append(options, fmt.Sprintf("verity.hashoffset=%d", opts.VerityHashOffset))
```

Comment:
Should this be conditional on non-zero offset?

Reply:
I also agree.

Reply:
Separated it from the rest and also tried to add more explicit error handling since all those options need to be specified together (similarly to how we try and throw an error if tmpfs and fsck are both specified).

Reply:
Shouldn't this move inside the `if len(opts.VerityHashDevice) > 0 && len(opts.VerityRootHash) > 0 `? I read this as that these two parameters need to exist for the offset to make sense.

Reply:
@alfonsosanchezbeato I moved it inside the first if. Yes that's correct the offset makes sense only if the other parameters are also specified. 

I might refactor this bit to group verity related opts (same for overlay ones) together as a new struct type and have it somehow checked in a nicer more compact way than all those ifs.